sdk/servicebus/Azure.Messaging.ServiceBus/src/Compatibility/ServiceBusAmqpExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Messaging.ServiceBus.Amqp;
+using Microsoft.Azure.Amqp;
+using AmqpLib = Microsoft.Azure.Amqp;
+
+namespace Azure.Messaging.ServiceBus.Primitives
+{
+    /// <summary>
+    /// Extensions for converting to and from AMQP.
+    /// </summary>
+    public static class ServiceBusAmqpExtensions
+    {
+        /// <summary>
+        /// Converts the <see cref="ServiceBusReceivedMessage"/> to its serialized AMQP form.
+        /// </summary>
+        /// <returns></returns>
+        public static BinaryData ToAmqpBytes(this ServiceBusReceivedMessage message)
+        {
+            var stream = AmqpMessageConverter.Default.AmqpAnnotatedMessageToAmqpMessage(message.GetRawAmqpMessage()).ToStream();
+            return BinaryData.FromStream(stream);
+        }
+
+        /// <summary>
+        /// Constructs a <see cref="ServiceBusReceivedMessage"/> from its serialized AMQP form.
+        /// </summary>
+        /// <param name="messageBytes">The AMQP message bytes.</param>
+        /// <param name="lockTokenBytes">The lock token bytes.</param>
+        /// <returns></returns>
+        public static ServiceBusReceivedMessage FromAmqpBytes(BinaryData messageBytes, BinaryData lockTokenBytes)
+        {
+            var bufferStream = BufferListStream.Create(messageBytes.ToStream(), 4096);
+            var amqpMessage = AmqpLib.AmqpMessage.CreateInputMessage(bufferStream);
+
+            var message = AmqpMessageConverter.Default.AmqpMessageToSBReceivedMessage(amqpMessage);
```

Comment:
This also looks like some unnecessary copies. Is it not possible to simply read a byte array as a SBReceivedMessage?

Reply:
No, it needs to be read into an AMQP message first as this is where all of the decoding logic lives.

Reply:
We might want to ask the AMQP team to add an API for it.

Reply:
Well the SBReceivedMessage is a Service Bus concept. Our service agnostic AMQP model is AmqpAnnotatedMessage, which lives in Azure.Core.Amqp, but does not contain any serialization logic.